### ISSUE TITLE
docker rm + prefix has wrong error message whith multiple matches

### ISSUE THREAD

**Issue by bmedici** [moby/moby#38345]
**Description**

When removing containers with the same ID prefix, and we have multiple matches, the error message complains about this but displays one random ID which is irrelevant.
May be related or duplicate of [#37904]

```
▶ docker rm -f dc
Error response from daemon: Multiple IDs found with provided prefix: dc59acf4794bc1ef5213f76553183f3fee42576f21af2fa05659960e9f534119
```

**Steps to reproduce the issue:**
1. create multiple containers sharing the same id prefix
2. docker rm -f dc
3. see that docker complains with the prefix matching multiple containers but displaying one of the containers ID

**Describe the results you received:**
```
▶ docker rm -f dc
Error response from daemon: Multiple IDs found with provided prefix: dc59acf4794bc1ef5213f76553183f3fee42576f21af2fa05659960e9f534119
```

**Describe the results you expected:**
```
▶ docker rm -f dc
Error response from daemon: Multiple IDs found with provided prefix "dc"
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client: Docker Engine - Community
 Version:           18.09.0
 API version:       1.39
 Go version:        go1.10.4
 Git commit:        4d60db4
 Built:             Wed Nov  7 00:47:43 2018
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.0
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.4
  Git commit:       4d60db4
  Built:            Wed Nov  7 00:55:00 2018
  OS/Arch:          linux/amd64
  Experimental:     true
```

**Output of `docker info`:**

```
Containers: 15
 Running: 0
 Paused: 0
 Stopped: 15
Images: 143
Server Version: 18.09.0
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 468a545b9edcd5932818eb9de8e72413e616e86e
runc version: 69663f0bd4b60df09991c08812a60108003fa340
init version: fec3683
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.125-linuxkit
Operating System: Docker for Mac
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.952GiB
Name: linuxkit-025000000001
ID: 63X6:E7RI:5ETX:6UIG:FLU2:7CJA:PUJS:R2Y6:MHZF:FJ2I:HCJ7:LR53
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 26
 Goroutines: 53
 System Time: 2018-12-11T10:43:49.1733366Z
 EventsListeners: 2
HTTP Proxy: gateway.docker.internal:3128
HTTPS Proxy: gateway.docker.internal:3129
Registry: https://index.docker.io/v1/
Labels:
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Product License: Community Engine
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by thaJeztah** [comment]
Looks to be the same issue; and looking at the code in the 18.09.0 tag, I see that fix is not yet in Docker 18.09; https://github.com/docker/engine/blob/v18.09.0/pkg/truncindex/truncindex.go#L111

I'll open a backport for that fix (not sure it will make it in time for 18.09.1)

**Comment by thaJeztah** [comment]
Opened https://github.com/docker/engine/pull/132 to backport the fix to 18.09.x

I'll close this issue because the issue has been fixed already on master (which is what's being tracked in this repository), but feel free to continue the conversation

**Comment by thaJeztah** [comment]
Verfied the patch on the backport PR;

```
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
e5dc4c91b45b        busybox             "sh"                35 seconds ago      Created                                 determined_cartwright
e46c4ed73d8c        busybox             "sh"                40 seconds ago      Created                                 nervous_varahamihira
```

```bash
 docker rm e
Error response from daemon: Multiple IDs found with provided prefix: e
```
